### PR TITLE
MM-47442 : Avoid passing global state into post_actions.ts in Mattermost webapp

### PR DESCRIPTION
**Pull request by pradeepmurugesan**
#### Summary
Avoid calling `getState()` as much as possible.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/21334

```release-note
NONE
```

### CODE REVIEW

File: actions/post_actions.ts [hunk 1]
```diff
@@ -278,12 +278,13 @@ export function markPostAsUnread(post: Post, location: string) {
...
 export function deleteAndRemovePost(post: Post) {
     return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        const state = getState();
         const {error} = await dispatch(PostActions.deletePost(post));
         if (error) {
             return {error};
         }
 
-        if (post.id === getSelectedPostId(getState() as GlobalState)) {
+        if (post.id === getSelectedPostId(state as GlobalState)) {
```
Comment (M-ZubairAhmed) [inline actions/post_actions.ts:287]: @mattermost/web-platform this is the type casting we keep getting in many places, as the shared redux store globalstate is different from the webapp specific global state. How can we improve this.

**Review by M-ZubairAhmed — APPROVED** [review]
💯 It looks good to me. Thank you 🎉

File: actions/post_actions.ts [hunk 2]
```diff
@@ -292,7 +293,7 @@ export function deleteAndRemovePost(post: Post) {
             });
         }
 
-        if (post.id === getSelectedPostCardId(getState() as GlobalState)) {
+        if (post.id === getSelectedPostCardId(state as GlobalState)) {
```
Comment (koox00) [inline actions/post_actions.ts:296]: We need to get the changed state after dispatching `SELECT_POST` so calling `getState()` again is needed in this case.

**Review by koox00 — CHANGES_REQUESTED** [review]

File: actions/post_actions.ts (hunk 2 above)
Comment (koox00) [reply actions/post_actions.ts:296]: hmm, my mistake, I didn't read the full block, it seems that the previous action won't affect the next one.

Still, I don't really get what we are trying to achieve here..

**Comment by AshishDhama** [comment]
@nevyangelova Could you please confirm if this is what we wanted to change?

**Comment by nevyangelova** [comment]
Sorry for the long wait on this, I updated the ticket and have been talking to @pradeepmurugesan in DM as well. We would ideally be calling getState as long as it's coming from a thunk inside the component. We don't want to import it from stores/redux_store

```Proposed change in post_actions.jsx:

// get the state from getState
  ......
  dispatch({
            type: SearchTypes.RECEIVED*SEARCH*POSTS,
            data: {posts, order: newResults},
        });
} 
...
dispatch(addPostToSearchResults(postId));
```

**Comment by AshishDhama** [comment]
@pradeepmurugesan Thanks for the effort.
closing the pr as changes aren’t required.

**Pull request closed without merging**